src/docfx/build/toc/ServicePageGenerator.cs

```diff
@@ -86,9 +86,10 @@ internal class ServicePageGenerator
                                 hrefFileFullPath = Path.GetFullPath(Path.Combine(topLevelTOCYmlDir == null ? "" : topLevelTOCYmlDir, childHref));
                             }
 
-                            var servicePageFullPath = Path.GetDirectoryName(Path.GetFullPath(Path.Combine(_docsetPath, servicePagePath.Path))) ?? _docsetPath;
-                            var hrefRelativePath = Path.GetRelativePath(servicePageFullPath, hrefFileFullPath);
-                            childHref = hrefRelativePath;
+                            var hrefRelativePathToDocset = Path.GetRelativePath(_docsetPath, hrefFileFullPath);
+                            var hrefFullPathOfDocset = $"~/{hrefRelativePathToDocset}";
```

Comment:
what if we do nothing here, and calculate the relative path based on the `original relative path`, `path of top level toc` and `path of reference toc`?

Reply:
Finished.

Reply:
Fixed/